### PR TITLE
Added call to Llamahub proxy to track download numbers

### PR DESCRIPTION
**Pull request by pankhudib1234**
# Description
Added tracking code in following functions
* download_loader
* download_tool
* download_llama_pack
* download_lllama_datasets

Fixes # (issue)
NA 

## Type of Change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Using python shell. Verified that Llama proxy is hit successfully & download count is updated for each of the four plugin types. 

# Todo
How can we track the downloads when download-utils is not called, i.e., when users directly imports a package.

### CODE REVIEW

File: llama_index/download/download_utils.py [hunk 1]
```diff
@@ -26,6 +27,12 @@
 
 PATH_TYPE = Union[str, Path]
 
+LLAMAHUB_ANALYTICS_PROXY_SERVER = "https://llamahub.ai/api/analytics/download"
+class MODULE_TYPE(str, Enum):
+    LOADER = "loader"
+    AGENT = "agent"
```
Comment (Disiok) [inline llama_index/download/download_utils.py:33]: I think technically it's tools instead of agents.

**Review by Disiok — APPROVED** [review]
nice, minor nits.

File: llama_index/download/download_utils.py (hunk 1 above)
Comment (pankhudib1234) [reply llama_index/download/download_utils.py]: changed to tool

File: llama_index/download/download_utils.py [hunk 2]
```diff
@@ -345,3 +352,17 @@ def download_llama_module(
...
+) -> None:
+    """Tracks number of downloads via Llamahub proxy
+    Args:
+        module_class: The name of the llama module being downloaded, e.g.,`GmailOpenAIAgentPack`.
+        module_type: Can be "loader", "tool", "llamapack", or "datasets"
+    """
+    try: 
+        requests.post(LLAMAHUB_ANALYTICS_PROXY_SERVER, json={"type": module_type, "plugin": module_class})
+    except: 
+        print(f"Error tracking downloads for {module_class}")
```
[the remark this answers is not in the document]
Comment (pankhudib1234) [reply llama_index/download/download_utils.py]: switched to logger.

**Pull request merged**